src/catalog/tests/open.rs

```diff
@@ -764,3 +769,88 @@ async fn test_tombstone(
 
     assert_eq!(openable_state2.get_tombstone().await.unwrap(), Some(true));
 }
+
+#[mz_ore::test(tokio::test)]
+#[cfg_attr(miri, ignore)] //  unsupported operation: can't call foreign function `TLS_client_method` on OS `linux`
+async fn test_persist_fencing() {
+    async fn new_persist_client(version: semver::Version) -> PersistClient {
+        let mut cfg = PersistConfig::new_for_tests();
+        cfg.build_version = version;
+        let metrics = &MetricsRegistry::new();
+        let mut cache =
+            PersistClientCache::new(cfg, metrics, |_, _| PubSubClientConnection::noop());
+        cache
+            .open(PersistLocation::new_in_mem())
+            .await
+            .expect("in-mem location is valid")
+    }
```

Comment:
@danhhz Can I get a little help writing this test? Every time I call this method it seems to wipe the old state and create a brand new one. What I'm trying to do is open the existing state with a new version.